app.js

```diff
@@ -12,7 +13,84 @@ app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body['access_token']))
+  .catch((error) =>
+    console.log('Something went wrong when retrieving an access token', error)
+  );
+
+//Middleware for parsing JSON and form-data requests
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
 
 // Our routes go here:
+//Homepage
+app.get('/', (req, res) => {
+  res.render('index');
+});
+
+// GET /search?...
+app.get('/artist-search', (req, res) => {
+  console.log(`req.query`, req.query); //URL query values
+  const searchedArtist = req.query.artistName; //Grab the value from the query
+  console.log(searchedArtist);
+
+  //Find the artists with searched name
+  spotifyApi
+    .searchArtists(searchedArtist)
+    .then((data) => {
+      console.log('The received data from the API: ', data.body.artists.items);
+      //console.log(data.body.artists.items[0].images);
+      // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+      res.render('artist-search-results', { artists: data.body.artists.items });
```

Comment:
you can save _data.body.artists.items_ in a variable before passing it to the view, just for making your code cleaner ;)